cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/spark/SparkRuntimeService.java

```diff
@@ -414,6 +432,31 @@ private File saveCConf(CConfiguration cConf, File targetDir) throws IOException
   }
 
   /**
+   * Creates a jar in the given directory that contains a logback.xml loaded from the current ClassLoader.
+   *
+   * @param targetDir directory where the logback.xml should be copied to
+   * @return the {@link File} where the logback.xml jar copied to or {@code null} if "logback.xml" is not found
+   *         in the current ClassLoader.
+   */
+  @Nullable
```

Comment:
It is a bit similar to MR, but not exactly the same. We can abstract out the core piece (the finding "logback.xml" from a classloader and create a jar to a output supplier) and call them with different classloader and output supplier, which I need to find a place to host that method (maybe some util class, but we don't have something similar yet). I am open to do the refactoring.